crypto/asn1/standard_methods.h

```diff
@@ -59,3 +59,61 @@ static const EVP_PKEY_ASN1_METHOD *standard_methods[] = {
 #endif
 };
 
+static const EVP_PKEY_ASN1_METHOD *standard_methods_fips[] = {
```

Comment:
is this necessary to break out into its own?
We don't plan on releasing this outside of FIPS right and even if we did we would not compile with the FIPS options so they would still work out right ?

If the is all true the we could eliminate the changes to `crypto/asn1/ameth_lib.c**crypto/asn1/ameth_lib.c`

If there is an additional detail I'm missing that would be helpful, thanks

Reply:
The reason I added it to crypto/asn1/standard_methods.h instead of adding it to the .c file was that it is *much* easier to confirm that the array added here is identical to the "standard_methods" array with the addition of the #ifdefs around FIPS to remove the unwanted algos.

If it was in the .c file, then it would be much easier for a subsequent CVE patch to maybe change the "standard_methods" array in the .h file, and for us to forget to make the same change in the "standard_methods_fips array" at the same time.

Keeping the two arrays together makes it easier to keep them in sync. This is just my experience of many years of code changes like this :-).

Reply:
I see you point, we'd just need to make sure that if we're addressing CVEs and it modifies the array[] `static const EVP_PKEY_ASN1_METHOD *standard_methods` that we also replicate the changes to `static const EVP_PKEY_ASN1_METHOD *standard_methods_fips` 

I think maybe that would have been a case to integrate the `#ifndef FIPS` into the `static const EVP_PKEY_ASN1_METHOD *standard_methods` it would conflict and force us to make sure we know who to integrate the changes. 
Then we could also remove the `if else` in the `.c`.

If this is a method you're comfortable with maintaining then I will not make a fuss.

Reply:
I just realized I didn't answer the first question: "is this necessary to break out into its own?"

Yes is has to be a separate array (with the changes to crypto/asn1/ameth_lib.c**crypto/asn1/ameth_lib.c also). The reason is that non-FIPS mode still has to work with this library, as this is how all the tests are run in the rpm build.

I originally tried to use only a truncated array, #ifdef'ed out, and this caused all the tests in the rpm build to fail.

So we need the original array for non-fips mode (it still has to work in that mode), and the truncated array for when we switch into fips mode.

Reply:
Cool thanks for the clarification here